components/brave_sync/brave_sync_service_unittest.cc

```diff
@@ -1259,3 +1275,277 @@ TEST_F(BraveSyncServiceTest, DeviceIdV2MigrationDupDeviceId) {
+    GenerateObjectIdForOtherNode(std::string());
+  auto record1 = SimpleFolderSyncRecord(
+      SyncRecord::Action::A_CREATE, object_id_iter1,
+      GetOtherNodeName(), kOtherNodeOrder, "", "", false, GetOtherNodeName());
+  EXPECT_TRUE(sync_service()->IsOtherBookmarksFolder(record1.get()));
+
+  bool pass_to_syncer = false;
+  sync_service()->ProcessOtherBookmarksFolder(record1.get(), &pass_to_syncer);
+  EXPECT_FALSE(pass_to_syncer);
+  std::string other_node_object_id;
+  model()->other_node()->GetMetaInfo("object_id", &other_node_object_id);
+  EXPECT_EQ(other_node_object_id, object_id_iter1);
+
+  const std::string object_id_iter2 =
+    GenerateObjectIdForOtherNode(object_id_iter1);
+  // Now we emulate our object id is out-dated, need to catch up with current
+  // one
+  auto record2 = SimpleFolderSyncRecord(
+      SyncRecord::Action::A_CREATE, object_id_iter2,
+      GetOtherNodeName(), kOtherNodeOrder, "", "", false, GetOtherNodeName());
+  EXPECT_TRUE(sync_service()->IsOtherBookmarksFolder(record2.get()));
+
+  pass_to_syncer = false;
+  sync_service()->ProcessOtherBookmarksFolder(record2.get(), &pass_to_syncer);
+  EXPECT_FALSE(pass_to_syncer);
+  other_node_object_id = "";
+  model()->other_node()->GetMetaInfo("object_id", &other_node_object_id);
+  EXPECT_EQ(other_node_object_id, object_id_iter2);
+
+  // Prepare children of other_node()
+  const bookmarks::BookmarkNode* folder_a =
+    model()->AddFolder(model()->other_node(), 0, base::ASCIIToUTF16("A"));
+  const bookmarks::BookmarkNode* bookmark_a1 =
+    model()->AddURL(model()->other_node(), 1, base::ASCIIToUTF16("A1"),
+                    GURL("https://a1.com"));
+  const std::string folder_a_object_id = GenerateObjectId();
+  const std::string bookmark_a1_object_id = GenerateObjectId();
+  AsMutable(folder_a)->SetMetaInfo("object_id", folder_a_object_id);
+  AsMutable(bookmark_a1)->SetMetaInfo("object_id", bookmark_a1_object_id);
+  AsMutable(folder_a)->SetMetaInfo("order",
+                                   std::string(kOtherNodeOrder) + ".1");
+  AsMutable(bookmark_a1)->SetMetaInfo("order",
+                                      std::string(kOtherNodeOrder) + ".2");
+  AsMutable(folder_a)
+    ->SetMetaInfo("sync_timestamp",
+                  std::to_string(base::Time::Now().ToJsTime()));
+  AsMutable(bookmark_a1)
+    ->SetMetaInfo("sync_timestamp",
+                  std::to_string(base::Time::Now().ToJsTime()));
+
+  // Prepare a folder for "Other Bookmarks" to move to
+  const bookmarks::BookmarkNode* folder_b =
+    model()->AddFolder(model()->bookmark_bar_node(), 0,
+                       base::ASCIIToUTF16("B"));
+  const std::string folder_b_object_id = GenerateObjectId();
+  AsMutable(folder_b)->SetMetaInfo("object_id", folder_b_object_id);
+  AsMutable(folder_b)->SetMetaInfo("order", "1.0.1.1");
+  AsMutable(folder_b)
+    ->SetMetaInfo("sync_timestamp",
+                  std::to_string(base::Time::Now().ToJsTime()));
```

Comment:
Here `folder_b` seems created in `Bookmarks bar` and have filled the metadata , but is not used below.
Why do we need `folder_b`?